test/verify/main.cpp

```diff
@@ -89,5 +89,6 @@ int main(int argc, const char* argv[])
                          "batch_quant_dot_1<migraphx::fp8::fp8e4m3fnuz, float>",
                          "quant_dot_3args_4<migraphx::fp8::fp8e4m3fnuz, float>",
                          "quant_dot_3args_5<migraphx::fp8::fp8e4m3fnuz, float>"});
+    rv.change_tolerance_for("test_quantizelinear_convert", 100000);
```

Comment:
Tolerance should be a method on the `verify_program` class. And then `test_quantizelinear_convert` can override it. We can add a `tolerance` member variable to the `program_info` class to store the result of the method.

Reply:
And does the `tolerance` need to be that high?

Reply:
> And does the tolerance need to be that high? 

Yes, test_verify takes floating point epsilon based on final output. Which is ~1e-7, RMS Error is ~0.01. Therefore need tolerance for 1e5. 

```
FAILED: cpu
RMS Error: 0.0104129
ref:2.75, 1.75, -1.75, 2.5, -1, 1, 0.625, 9
target:2.6875, 1.75, -1.75, 2.5, -1.0625, 1.0625, 0.625, 9.25
Max diff: 0.25
Mismatch at 0: 2.75 != 2.6875
```

Mismatch between numbers are coming due to the fact that For the "ref" implementation it downcasts to "fp8" first and then upcasts to float. 

For the CPU and GPU it optimizes away this converts. 

e.g. 

```
ref:
module: "main"
y_zero_point = @param:y_zero_point -> fp8e4m3fnuz_type, {2, 2, 2}, {4, 2, 1}, target_id=0
y_scale = @param:y_scale -> float_type, {2, 2, 2}, {4, 2, 1}, target_id=0
x = @param:x -> float_type, {2, 2, 2}, {4, 2, 1}, target_id=0
@3 = ref::quantizelinear(x,y_scale,y_zero_point) -> fp8e4m3fnuz_type, {2, 2, 2}, {4, 2, 1}, target_id=0
@4 = ref::convert[target_type=2](@3) -> float_type, {2, 2, 2}, {4, 2, 1}, target_id=0
@5 = @return(@4), target_id=0
```

for the GPU it is : 
```
template<class Tx0, class Tx1, class Tx2>
__device__ __attribute__((const)) auto inner_pointwise(Tx0 x0,Tx1 x1,Tx2 x2) {
// @literal -> float_type, {1}, {0}
auto z0 = float(240);
// @literal -> float_type, {1}, {0}
auto z1 = float(-240);
// @param:x2 -> fp8e4m3fnuz_type, {1}, {0}
// @param:x1 -> float_type, {1}, {0}
// @param:x0 -> float_type, {1}, {0}
// div -> float_type, {1}, {0}
auto z5 = migraphx::convert<float>(x0 / x1);
// nearbyint -> float_type, {1}, {0}
auto z6 = migraphx::convert<float>(migraphx::nearbyint(z5));
// convert[target_type=2] -> float_type, {1}, {0}
auto z7 = migraphx::convert<float>(migraphx::convert<float>(x2));
// add -> float_type, {1}, {0}
auto z8 = migraphx::convert<float>(z6 + z7);
// clip -> float_type, {1}, {0}
auto z9 = migraphx::convert<float>(migraphx::min(migraphx::max(z1, z8), z0));
// @return -> float_type, {1}, {0}
return z9;

}


extern "C" {
MIGRAPHX_GLOBAL void quantizelinear_convert_kernel(void * private_p0,void * private_p1,void * private_p2,void * private_p3)
{
    auto idx = make_index();
    pointwise(idx, vectorize<1, 0>())(MIGRAPHX_LIFT(inner_pointwise), private_p0,private_p1,private_p2,private_p3);
}

}

```

On the sidenote : now that we are calling `optimize_module`, naming of JIT kernel is not accurate. e.g. For this example it is `quantizelinear_convert` but `convert` is optimized out.